src/smart-pointers/rc.md

```diff
@@ -19,6 +19,22 @@ fn main() {
 }
 ```
 
+Each `Rc` points to the same shared data structure, containing strong and weak
+pointers and the value:
+
+```bob
+ Stack                     Heap
+.- - - - - - - -.     .- - - - - - - - - - - - - - - - -.
+:               :     :                                 :
+:     +-----+   :     :   +-------------+-----------+   :
+:  a: | o---|---:-----:-->|  strong: 2  |  weak: 0  |   :
```

Comment:
Why not make it `weak: 2` and say that every strong reference is also considered a weak reference. Or just leave this slight "lie". I don't think people taking this course need this level of detail to be correct.

Reply:
I just omitted the "weak" bit -- that can remain an implementation detail :)